### PR TITLE
Use pushgateway for reporting progress/success on hathifiles

### PR DESCRIPTION
**Pull request by aelkiss**
This needs https://github.com/hathitrust/push_metrics/pull/2 merged before taking this out of draft (and using the `main` branch again for pushmetrics).

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -4,15 +4,14 @@ services:
 
   hf:
     build: .
-    ports:
-      - "4567:4567"
```
Comment (mwarin) [inline docker-compose.yml:8]: Do we know of any external dependencies on these ports for this service?

File: jobs/update_hathifile_listing.rb [hunk 2]
```diff
@@ -49,6 +50,7 @@ def run
 
     file_list_file = File.open(hathifile_listing, "w")
     file_list_file.puts file_list.sort_by { |hf| hf["filename"] }.to_json
+    tracker.log_final_line
```
Comment (mwarin) [inline jobs/update_hathifile_listing.rb:53]: No tracker.increment in this script? Can't tell if omitted intentionally or un-.

File: spec/jobs/generate_hathifile_spec.rb [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 require "spec_helper"
 require "zlib"
+require "faraday"
```
Comment (mwarin) [inline spec/jobs/generate_hathifile_spec.rb:5]: I'm not familiar with faraday and don't understand why we're using it here. 
I assume a stdlib Net::HTTP is not sufficient?

**Review by mwarin — APPROVED** [review]
I see no reason to hold it back, just a few questions based on my not knowing stuff.

File: spec/jobs/generate_hathifile_spec.rb (hunk 3 above)
Comment (aelkiss) [reply spec/jobs/generate_hathifile_spec.rb:5]: Net::HTTP probably would be fine for this case, but faraday is somewhat less verbose and (generally) the recommendation for an HTTP library to use.

File: docker-compose.yml (hunk 1 above)
Comment (aelkiss) [reply docker-compose.yml:8]: I think this was copy-pasta - it doesn't even start a service that runs on those ports.

File: jobs/update_hathifile_listing.rb (hunk 2 above)
Comment (aelkiss) [reply jobs/update_hathifile_listing.rb:53]: That's probably an omission, good catch - the test doesn't check for a nonzero number.  We could probably increment for each file in the `hathifile_web_dir` that we examine.

Comment (aelkiss) [reply jobs/update_hathifile_listing.rb:54]: Added.

**Comment by aelkiss** [comment]
I am going to go ahead and squash & merge.

**Pull request merged**